Add basic implementation of `Aqua Blast` mob skill

mrhappyasthma opened the pull request:
There was not a whole lot available about this spell online, so I implemented it to behave largely like `aqua_breath.lua`. Also, for most of the knockback, I just put a placeholder value.

Most knockback spells appear to use `1` for the value in the SQL, so I put that as the placeholder in the SQL for this spell as well.

(This doc also mentioned as a TODO for this spell, as well: https://github.com/LandSandBoat/server/blob/base/documentation/Mobskills_Knockback_Values_TODO.xml#L12)

This change is done as part of https://github.com/LandSandBoat/server/issues/201

<!-- remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm: -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

mrhappyasthma committed:
Add basic implementation of `Aqua Blast` mob skill

There was not a whole lot available about this spell online, so I implemented it to behave largely like `aqua_breath.lua`. Also, for most of the knockback, I just put a placeholder value.

Most knockback spells appear to use `1` for the value in the SQL, so I put that as the placeholder in the SQL for this spell as well.

(This doc also mentioned as a TODO for this spell, as well: https://github.com/LandSandBoat/server/blob/base/documentation/Mobskills_Knockback_Values_TODO.xml#L12)

This change is done as part of https://github.com/LandSandBoat/server/issues/201

mrhappyasthma committed:
Fix whitespace. (I really need to update my editor...)

Xaver-DaRed commented on scripts/globals/mobskills/aqua_blast.lua:
```diff
@@ -0,0 +1,33 @@
+
+mobskill_object.onMobWeaponSkill = function(target, mob, skill)
+    local dmgmod = MobBreathMove(mob, target, 0.25, 1.5, xi.magic.ele.WATER, 400)
+    local dmg = MobFinalAdjustments(dmgmod, mob, skill, target, xi.attackType.BREATH, xi.damageType.WATER, MOBPARAM_WIPE_SHADOWS)
```
xi.attackType.BREATH <-- Is it?
Searched some info:
https://nw6yx36onohv5j6wmzoba3nllq-ac4c6men2g7xr2a-wiki-ffo-jp.translate.goog/html/24704.html

i think it should be magical damage type

Adding to that, maybe its necesary to edit "MobBreathMove" aswell

mrhappyasthma replied:
Wow nice find.

Copying for reference from the translation of that link:

"The name means "water gust", and it looks like abreath- type attackthat spits out a stream of water from the mouth , but it seems to be a kind of magic damage because it can be defended inthe sanctuaryof the goddess ."

It also mentions "it has a knockbackeffect over a fairly long distance", so I will update the placeholder to 2 until we can get a packet catpure.

Xaver-DaRed replied:
I left a link to the mob family in the open issue.
There you can "easily" get some info on its mob skills

mrhappyasthma replied:
Very helpful. Thank you!

mrhappyasthma replied:
Applied those changes.

One open question: how to determine proper damage values for spells like these? It seems most of the spells using `MobMagicalMove` multiply the `mob:getWeaponDmg()` by some fixed constant (between 2-8) and leave `dmgmod = 1`.

But it's unclear if these are all just arbitrarily chosen, or come from some more definitive source.

I went with something that seemed reasonable given other spell(s) that are similar. But right now it's largely just a "magic number".

Xaver-DaRed replied:
Ok so... Taking the words of certain wise man...
As you have probably imagined by now, damage calculations are NOT being handled as they should.

Some skills have "random" values to try to come close to what retail does with our calculations.
Others use the values from wiki or other sources and leave it as is.

The whole system needs a rework. In my book, both ways are ok for now, so long as you state what method you have chosen.

mrhappyasthma replied:
My process wasn't very scientific.

I tried to pick a reasonable median value from a bunch of similar Water magic spells that use the same constant for the damage. E.g. `regurgitation`, `acid_spray`, `water_ii`, `gastric_bomb`, `aqua_ball`, and more.

mrhappyasthma committed:
Update range and knockback.

This better matches the japanese wiki, which has more info:
https://nw6yx36onohv5j6wmzoba3nllq-ac4c6men2g7xr2a-wiki-ffo-jp.translate.goog/html/24704.html

mrhappyasthma committed:
Change from breath to Water magic

"The name means "water gust", and it looks like abreath- type attackthat spits out a stream of water from the mouth , but it seems to be a kind of magic damage because it can be defended inthe sanctuaryof the goddess ."

https://nw6yx36onohv5j6wmzoba3nllq-ac4c6men2g7xr2a-wiki-ffo-jp.translate.goog/html/24704.html

Xaver-DaRed commented:
Rebase/resolve conflicts
Other than that, looks good to me

mrhappyasthma commented:
my rebase went wild. Please ignore this. will fix and re-push

mrhappyasthma commented:
Corrected.

mrhappyasthma commented:
It shows a Style_Checks failing now. Will see what happened.

mrhappyasthma commented:
Error shows:

```
Checking scripts/globals/mobskills/aqua_blast.lua 4 warnings

    scripts/globals/mobskills/aqua_blast.lua:27:18: (W113) accessing undefined variable MobMagicalMove
    scripts/globals/mobskills/aqua_blast.lua:27:105: (W113) accessing undefined variable TP_NO_EFFECT
    scripts/globals/mobskills/aqua_blast.lua:28:17: (W113) accessing undefined variable MobFinalAdjustments
    scripts/globals/mobskills/aqua_blast.lua:28:111: (W113) accessing undefined variable MOBPARAM_WIPE_SHADOWS
```

Not exactly sure why. All 4 come from this require: `require("scripts/globals/monstertpmoves")`.

If anyone sees something I'm missing, please let me know.

mrhappyasthma commented:
Shows also `[29/May] [05:13:41] [Error]  make_listen_bind: bind failed (socket #5, code 98)!`.

However testing on my local server still shows this working.

Xaver-DaRed commented:
Ignoring both fails.
the style checks are probably handled already (read:ignored)
the other is a listener to socket fail, wich, happen
